src/IonTextWriter.ts

```diff
@@ -226,6 +226,9 @@ export class TextWriter extends AbstractWriter {
     }
 
     stepIn(type: IonType) : void {
+        if (this.currentContainer.containerType === IonTypes.STRUCT && this.currentContainer.state === State.STRUCT_FIELD) {
```

Comment:
Isn't this technically redundant as `this.currentContainer.state === State.STRUCT_FIELD` implies `this.currentContainer.containerType === IonTypes.STRUCT`?